toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -493,6 +495,22 @@ func PopulateInstallRoot(installChroot *safechroot.Chroot, packagesToInstall []s
 		return
 	}
 
+	// imageconfigvalidator should have ensured that we intend to install shadow-utils, so we can go ahead and do that here.
+	if len(config.Users) > 0 || len(config.Groups) > 0 {
+		if !sliceutils.ContainsValue(packagesToInstall, "shadow-utils") {
+			err = fmt.Errorf("shadow-utils package is required when setting users or groups")
+			return
```

Comment:
Doesn't this belong in the config validation code earlier on?

Reply:
We put as much validation as possible natively as part of each config structure. The validator tool will use these functions when it runs its validation, but the config validator's job is to check against external things that aren't encoded in the .json. All the user bits are part of the .json. This is a bit odd since its now just a warning while it was originally an actual error.